install.sh

```diff
@@ -94,133 +104,125 @@ esac
-  version=`echo $rest | sed -e 's/-chromeos.*$//'`
+for i in $(seq 0 $((${#urls[@]} - 1))); do
+  url="${urls["${i}"]}"
+  sha256="${sha256s["${i}"]}"
+  tarfile="$(basename ${url})"
+  name="${tarfile%%-*}"   # extract string before first '-'
+  rest="${tarfile#*-}"    # extract string after first '-'
+  version="$(echo ${rest} | sed -e 's/-chromeos.*$//')"
                         # extract string between first '-' and "-chromeos"
 
-  download_check $name $url $tarfile $sha256
-  extract_install $name $tarfile
-  update_device_json $name $version
+  download_check "${name}" "${url}" "${tarfile}" "${sha256}"
+  extract_install "${name}" "${tarfile}"
+  update_device_json "${name}" "${version}"
 done
 
-# download, prepare and install chromebrew
-cd $CREW_LIB_PATH
-rm -rf crew lib packages
-curl -# -o crew $URL/crew
-chmod +x crew
-rm -f $CREW_PREFIX/bin/crew
-ln -s `pwd`/crew $CREW_PREFIX/bin
-#install crew library
-mkdir -p $CREW_LIB_PATH/lib
-cd $CREW_LIB_PATH/lib
-curl -# -o package.rb $URL/lib/package.rb
-curl -# -o package_helpers.rb $URL/lib/package_helpers.rb
-
-# Making GCC act like CC (For some npm packages out there, only required for gcc)
-#rm -f $CREW_PREFIX/bin/cc
-#ln -s $CREW_PREFIX/bin/gcc $CREW_PREFIX/bin/cc
-
-# package gcc7 has already made symbolic links for cc and gcc, no action required here
+# create symlink to 'crew' in ${CREW_PREFIX}/bin/
+rm -f "${CREW_PREFIX}/bin/crew"
+ln -s "../lib/crew/crew" "${CREW_PREFIX}/bin/"
 
 # prepare sparse checkout .rb packages directory and do it
-cd $CREW_LIB_PATH
+cd "${CREW_LIB_PATH}"
 rm -rf .git
 git init
-git remote add -f origin https://github.com/$OWNER/$REPO.git
+git remote add -f origin "https://github.com/${OWNER}/${REPO}.git"
 git config core.sparsecheckout true
 echo packages >> .git/info/sparse-checkout
 echo lib >> .git/info/sparse-checkout
 echo crew >> .git/info/sparse-checkout
-git fetch origin master
-git reset --hard origin/master
+git fetch origin "${BRANCH}"
+git reset --hard origin/"${BRANCH}"
 
 # install a base set of essential packages
 yes | crew install buildessential less most
 
+echo
+echo "If you have installed Chromebrew in a directory other than '/usr/local',"
+echo "please run these commands:"
+echo "echo 'export CREW_PREFIX=${CREW_PREFIX}' >> ~/.bashrc && . ~/.bashrc"
+echo "echo 'export PATH=\"\${CREW_PREFIX}/bin:\${CREW_PREFIX}/sbin:\${PATH}\"' >> ~/.bashrc && . ~/.bashrc"
+echo "echo 'export LD_LIBRARY_PATH=${CREW_PREFIX}/lib${LIB_SUFFIX}' >> ~/.bashrc && . ~/.bashrc"
 echo
 echo "To set the default PAGER environment variable to be able to use less:"
-echo "echo \"export PAGER=$CREW_PREFIX/bin/less\" >> ~/.bashrc && . ~/.bashrc"
+echo "echo \"export PAGER='less'\" >> ~/.bashrc && . ~/.bashrc"
 echo
 echo "Alternatively, you could use most.  Why settle for less, right?"
-echo "echo \"export PAGER=$CREW_PREFIX/bin/most\" >> ~/.bashrc && . ~/.bashrc"
+echo "echo \"export PAGER='most'\" >> ~/.bashrc && . ~/.bashrc"
```

Comment:
Yes.  First of all, users can change their `PATH` from the default and secondly, if ChromeOS decides to add one of these commands in future releases, this assures the desired command is executed.

Reply:
`/usr/local/bin` is first in `${PATH}`. That guarantees that the locally-installed copy will be used first.

Reply:
If they want to change their `${PATH}`, let them do so. I'm not stopping them.

Reply:
If Chrome OS does choose to add one of these commands, it shouldn't make a difference. `most` is `most` and `less` is `less`.

Reply:
Except in cases where the system version of less/most in future releases is not updated like, hopefully, the Chromebrew version.

Reply:
Keep in mind that this is all speculative. What use would a browser-based OS have for a pager?

Reply:
True, but still: Why would Chrome OS *include* a pager if it's only a browser? They're probable not going to add one. Anyway, like I said before, Chromebrew will almost certainly be in front of the `PATH` if people follow instructions.

Reply:
Oh, and by the way, `EDITOR` is set to `/bin/nano`, which just happens to be nonexistent. I'm reverting this.

Reply:
I change my PATH all the time and sometimes I add stuff in the front so it executes those things first.  The point is you shouldn't assume the PATH is set a certain way since it can be changed by the user at any time.

Reply:
Yes, and if a user chooses to, they should also have the right to change their pager. These are just *suggested defaults*.

Reply:
Suggested defaults that were changed because you wanted to change them when they didn't need changed.

Reply:
Yes, but it's not just that I want to change them. Users could change their `PATH` and yet their `PAGER` would stay the same. What if they removed their `CREW_PREFIX` but installed Chromebrew in another directory and added that to their `PATH` first?